app/api/v2/identity/sessions.rb

```diff
@@ -136,6 +136,87 @@ def get_user(email)
             error!({ errors: ['identity.session.auth0.invalid_params'] }, 422)
           end
         end
+
+        namespace :switch do
+          desc 'Switch user session',
+               success: { code: 200, message: 'Session was switched' },
+               failure: [
+                 { code: 404, message: 'Record is not found' }
+               ]
+          params do
+            optional :oid,
+                     type: String,
+                     desc: 'Organization OID'
+          end
+          post do
+            user = User.find_by(uid: session[:uid])
+            error!({ errors: ['identity.session.not_found'] }, 404) unless user
+
+            aid = params[:oid]
+            unless aid.nil?
+              # Check account in the organization that user belong to
+              members = Membership.joins('LEFT JOIN organizations ON organizations.id = memberships.organization_id')
+                                  .where(user_id: user.id)
+                                  .select('memberships.*,organizations.name, organizations.organization_id')
+                                  .pluck(:organization_id, :'organizations.name', :'organizations.organization_id')
+                                  .map { |id, name, pid| { id: id, name: name, pid: pid } }
+              error!({ errors: ['identity.member.not_found'] }, 404) if members.nil? || members.length.zero?
+
+              # Check user is barong organization admin or not
+              if members.first[:id].zero?
+                # User is barong organization admin
+                oids = Organization.all.pluck(:id)
+              else
+                # User is organizationn admin/account
+                oids = Organization.where(id: members.pluck(:id)).pluck(:id)
```

Comment:
To be refactored, you can get the current user membership and check the role.

Reply:
Right now. I determine the level of user by check the membership relation that I think it make sense to organize the user in the organization. I'm worry about If we determine user permission/access from `role` column in the membership table the organization access level will not correct.

Reply:
> To be refactored, you can get the current user membership and check the role.

I got you. Right now I integrated the user role with abilities.

To verify admin with Organization ability

```ruby
admin_organization? :read, Organization
```

To restrict the access to Organization

```ruby
admin_organization_authorize! :read, Organization
```